From d5f49da0680a81a481d786976cdf1d99d7684e39 Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@caviumnetworks.com>
Date: Fri, 23 Feb 2018 14:59:31 -0800
Subject: [PATCH 0759/1239] ahci: update cmd_tbl and rx_fis to 64-bit

existing u32 datatypes may fail to use correct value
when mem pointer crosses 32-bit boundary.

Signed-off-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 include/ahci.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/ahci.h b/include/ahci.h
index fb96dd8861..4b2762ac28 100644
--- a/include/ahci.h
+++ b/include/ahci.h
@@ -137,8 +137,8 @@ struct ahci_ioports {
 	void __iomem	*port_mmio;
 	struct ahci_cmd_hdr	*cmd_slot;
 	struct ahci_sg		*cmd_tbl_sg;
-	ulong	cmd_tbl;
-	u32	rx_fis;
+	u64	cmd_tbl;
+	u64	rx_fis;
 };
 
 /**
-- 
2.29.0

